spring-cloud-aws-sns/src/main/java/io/awspring/cloud/sns/core/ListTopicArnResolver.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2013-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sns.core;
+
+import java.util.Optional;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.ListTopicsRequest;
+import software.amazon.awssdk.services.sns.model.ListTopicsResponse;
+import software.amazon.awssdk.services.sns.model.Topic;
+
+/**
+ * Basic implementation for resolving ARN from topicName. It is using {@link SnsClient#listTopics()} to determine topicArn for topicName.
+ *
+ * @author Matej Nedić
+ * @since 3.0.0
+ */
+public class ListTopicArnResolver implements TopicArnResolver {
```

Comment:
Also something does not feel right about this name. Maybe `TopicsListingTopicArnResolver`?